File: C++/CNOTE_Chef-and-Notebooks.c
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int testCase;
    cin >> testCase;

    while(testCase--){
        int x, y, k, n;
        cin >> x >> y >> k >> n;

        vector <int> page(n);
        vector <int> cost(n);

        for(int i = 0; i < n; i++){
            cin >> page[i] >> cost[i];
        }

        vector <int> data(n, 0);
        for(int i = 0; i < n; i++){
            if((( y+page[i]) >= x) && (k >= cost[i])){
                data[i] = 1;
            }
        }

        int c = 0;
        for(int i = 0; i < n; i++){
            if(data[i] != 1){
                c++;
            }
        }

        if(c == n)
            cout << "UnluckyChef" <<endl;
        else
            cout << "LuckyChef" <<endl;
    }
    return 0;
}

File: Java/CNOTE_Chef-and-Notebooks.java
import java.util.Scanner;

class ChefAndNotebooks {

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int testCase = scanner.nextInt();
		while (testCase-- != 0) {
			int poemLength = scanner.nextInt();
			int pagesLeft = scanner.nextInt();
			int moneyInHand = scanner.nextInt();
			int noOfNoteBookAvaliable = scanner.nextInt();

			int pagesNeeded = poemLength - pagesLeft;
			int[] pagesInNotebook = new int[noOfNoteBookAvaliable];
			int[] costOfNotebook = new int[noOfNoteBookAvaliable];

			for (int i = 0; i < noOfNoteBookAvaliable; i++) {
				pagesInNotebook[i] = scanner.nextInt();
				costOfNotebook[i] = scanner.nextInt();
			}

			int flag = 0;
			for (int i = 0; i < noOfNoteBookAvaliable; i++) {
				if ((pagesInNotebook[i] >= pagesNeeded) && (costOfNotebook[i] <= moneyInHand)) {
					flag++;
				}
			}
			if (flag == 0)
				System.out.println("UnluckyChef");
			else
				System.out.println("LuckyChef");
		}
	}

}
